core/src/main/java/com/amplifyframework/analytics/AnalyticsUserProfile.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics;
+
+/**
+ * Represents user specific data such as name, email, plan, location etc.
+ */
+public final class AnalyticsUserProfile {
+    private String name;
+    private String email;
+    private String plan;
+    private Location location;
+    private Properties customProperties;
+
+    public AnalyticsUserProfile(final String name,
+                         final String email,
+                         final String plan,
+                         final Location location,
+                         final Properties customProperties) {
+        this.name = name;
+        this.email = email;
+        this.plan = plan;
+        this.location = location;
+        this.customProperties = customProperties;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getPlan() {
+        return plan;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public Properties getCustomProperties() {
+        return customProperties;
+    }
+
+    public static final class Location {
+        private Double latitude;
+        private Double longitude;
+        private String postalCode;
+        private String city;
+        private String region;
+        private String country;
+
+        public Location(final Double latitude,
```

Comment:
Important: add `equals(...)`, `toString()`, and `hashCode()` for this `Location` class.

How many of these fields are optional? Can you add `@Nullable` to getters and constructor params for any that aren't required? And `@NonNull` for those that must exist?

If `latitude` == `null` means "it wasn't provided", that would be a good thing to document also. Otherwise, use just the primitive `double` if your location class will always have lat/long.

The ideal situation is that all fields are `@NonNull`, and you use primitive `double`s. This way, there's no guess work for the user. If a `Location` was able to be constructed, that means its fields are populated, and the user can just use it without fear of `NullPointerException`: `location.getRegion()`, etc.

Reply:
Added null notations where ever applicable.